integration-cli/docker_api_swarm_test.go

```diff
@@ -321,9 +321,14 @@ func (s *DockerSwarmSuite) TestAPISwarmLeaderElection(c *testing.T) {
 			// clear these out before each run
 			leader = nil
 			followers = nil
+			type timeoutError interface{ Timeout() bool }
 			for _, d := range nodes {
 				n := d.GetNode(c, d.NodeID(), func(err error) bool {
-					if strings.Contains(errors.Cause(err).Error(), context.DeadlineExceeded.Error()) || strings.Contains(err.Error(), "swarm does not have a leader") {
+					if e, ok := errors.Cause(err).(timeoutError); ok && e.Timeout() {
+						lastErr = err
+						return true
+					}
+					if strings.Contains(err.Error(), "swarm does not have a leader") {
```

Comment:
I see a lot of similar checks to this scattered about. It might be helpful to unify them into an error checking function somewhere to reduce the impact if this error message was ever changed by some innocent seeming PR to swarmkit.

This appears to be checking for `ErrLostQuorum`, which could also be exported from what I can see? Maybe in future a PR to swarmkit to export this would be better, then we could compare by value instead of a brittle string match.